Documentation/kubernetes.md

```diff
@@ -0,0 +1,114 @@
+# Kubernetes authentication through dex
+
+## Overview
+
+Token responses from OpenID Connect providers include a signed JWT called an ID Token. ID Tokens contain names, emails, unique identifiers, and in dex's case, a set of groups that can be used to identify the user. OpenID Connect providers, such as dex, publish public keys, and the Kubernetes API server understands how to use these to verify ID Tokens.
```

Comment:
Small flow suggestion:

----
[...] OpenID Connect providers, like dex, publish public keys; the Kubernetes API server understands how to use these public keys to verify ID Tokens.